src/codecs/format/json.rs

```diff
@@ -80,13 +84,60 @@ impl From<&JsonDeserializerConfig> for JsonDeserializer {
     }
 }
 
+/// Config used to build a `JsonSerializer`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct JsonSerializerConfig;
+
+impl JsonSerializerConfig {
+    /// Creates a new `JsonSerializerConfig`.
+    pub const fn new() -> Self {
+        Self
+    }
+}
+
+#[typetag::serde(name = "json")]
+impl SerializerConfig for JsonSerializerConfig {
+    fn build(&self) -> crate::Result<BoxedSerializer> {
+        Ok(Box::new(JsonSerializer))
+    }
+}
+
+/// Serializer that converts an `Event` to bytes using the JSON format.
+#[derive(Debug, Clone)]
+pub struct JsonSerializer;
+
+impl JsonSerializer {
+    /// Creates a new `JsonSerializer`.
+    pub const fn new() -> Self {
+        Self
+    }
+}
+
+impl Encoder<Event> for JsonSerializer {
+    type Error = crate::Error;
+
+    fn encode(&mut self, event: Event, buffer: &mut bytes::BytesMut) -> Result<(), Self::Error> {
+        let json = match event {
+            Event::Log(log) => serde_json::to_vec(&log),
+            Event::Metric(metric) => serde_json::to_vec(&metric),
+        }?;
+
+        buffer.put(json.as_slice());
```

Comment:
Probably better to use `to_writer` and avoid the extra allocation + copy here.

Reply:
Didn't know about that API, very neat!

Reply:
Applied that suggestion in 650c17f4e61710b942b574687e491a6410dd5f88.